### DISCUSSION TITLE
What is the intent for the "-vhost-input" option?

### DISCUSSION THREAD

**Discussion by CyberStryk** [projectdiscovery/httpx#1472] [Q&A]
Does anyone know what the original idea for the -vhost-input httpx option was and/or is? 

I'm working an httpx feature to add vhost support (setting the "Host: CustomHost>" header in the request). The help menu for -vhost-input sounds to me like it should take a list of vhosts as an input and run those as vhosts against all the targets. Is this the intended functionality?

It's currently a scanoption as a boolean value and doesn't seem like it's used much in the current release. 

Looking for any feedback anyone can provide. If it turns out I have the right idea of what it's for I'll use it in my fork. If not I'll need to create a new option. 

Thanks for your help!

**Comment by DFC302** [comment]
From my understanding, the option is used to determine if the host you are scanning is using virtual hosting.

I am more interested in understanding how it determines valid vhosts, so easily.

But in response to your question, I don't think it's intended to take a list of vhosts as much as it's intended for you you to feed a list of hosts and use the option to determine which hosts are vhosts and which are not. So far, in my experience it has been fairly accurate. I hope that makes sense.

**Reply by CyberStryk** [reply]
Thanks for the response! 

I was under the impression that the "-vhost   - probe and display server supporting VHOST" option from the help is for the functionality you described above. That flag will show which servers are  potentially running vhost. Is that different than what you're saying -vhost-input does?

**Comment by intrd** [comment]
It seems that the `-vhost-input` option is still not working. The only method I found was using `-H "Host: vhost"`, but that only tests one vhost at a time, not a list, which I believe would be the intended function of this option.

**Comment by stefanovettorazzi** [comment]
I tried this feature and it doesn't work for me. 
Passing the `-debug` and `-verbose` flags doesn't show any progress.
I'm probably missing something, but I'm not sure. 
[It looks that it's implemented](https://github.com/projectdiscovery/httpx/blob/d58ad9d4c958d00394b6206a8074836117f20b56/runner/runner.go#L1492) but I get no errors and there are no requests made to the target.
The command I'm using is: `httpx -u ip_of_the_target -vhost-input list_of_vhosts.txt`.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.

**Comment by adilburaksen** [comment]
`-vhost-input` is for **virtual host probing**. You give it a wordlist of hostnames, and httpx cycles through them by setting the `Host` header to each value against your target IP/domain. It then compares each response against the baseline to flag vhosts that return something different.

```bash
httpx -u https://192.168.1.1 -vhost-input vhosts.txt
```

The reason it seems broken for a lot of people: if the server returns the same response for every `Host` value (which many do), nothing gets flagged — httpx needs a detectable difference in status, length, or title to mark something as a valid vhost.

It's essentially the same idea as `ffuf -H "Host: FUZZ"` with a comparison baseline baked in. If you need more control over the matching logic, the ffuf approach gives you more flexibility.
